docs/madr/decisions/070-resource-identifier.md

```diff
@@ -0,0 +1,332 @@
+
+#### What characters we can use in the identifier?
+ 
+In URL path segment we can use:
+
+```
+ALPHA / DIGIT / "-" / "." / "_" / "~" / "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "=" / ":" / "@"
+```
+
+In Envoy resource names we can use any character except `:` if we want to solve [Issue #2363](https://github.com/kumahq/kuma/issues/2363) without workaround.
+
+In Envoy stats fields we can use any character except `:` to avoid on the fly conversion of `:` to `_` for Cluster names in Prometheus labels.
+
+This leaves us with the following charset:
+
+```
+resource-identifier = *(ALPHA / DIGIT / "-" / "." / "_" / "~" / "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "=" / "@")
+```
+
+#### What characters we can use as a delimiter?
+
+As a delimiter we can use only characters that can't be present in resource identifier components – `name`, `mesh`, `namespace`, `zone`, `resourceType` and `sectionName`:
+
+* `name` and `zone`
+  * contain no more than 253 characters
+  * contain only lowercase alphanumeric characters, '-' or '.'
+  * start with an alphanumeric character
+  * end with an alphanumeric character
+* `mesh`
+  * contain no more than 63 characters
+  * contain only lowercase alphanumeric characters, '-' or '.'
+  * start with an alphanumeric character
+  * end with an alphanumeric character
+* `namespace`
+  * contain at most 63 characters
+  * contain only lowercase alphanumeric characters or '-'
+  * start with an alphanumeric character
+  * end with an alphanumeric character
+* `resourceType`
+  * only alphanumeric
+* `sectionName`
+  * contain at most 15 characters
+  * contain only lowercase alphanumeric characters or '-'
+
+This leaves us with the following charset:
+
+```
+delimiter = "_" / "~" / "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "=" / "@"
+```
+
+## Decision Drivers
+
+- New Inspect API endpoints that'd allow getting resolved configuration by the resource identifier
+- Kuma API GET endpoints shouldn't require core name to read the resource 
+- Human-readable: users should be able to type the identifier manually if needed, so it's not hashed
+- Envoy resources that have direct correlation with Kuma resources should be named by using resource identifier, i.e. outbound cluster can be named after MeshService
+
+## Considered Options
+
+* Option 1 - Order-based, no field names in the identifier
+* Option 2 - Field names in the identifier
+
+## Decision Outcome
+
+* Option 1 - Order-based, no field names in the identifier
+
+## Pros and Cons of the Options
+
+### Option 1 - Order-based, no field names in the identifier
+
+We need to pick a delimiter that's present in the `delimiter` charset we've specified above.
+
+There is an identifier format from Amazon called [ARN](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html). We can adopt a similar approach, but using `_`:
+
+```
+kri_<mesh>_<zone>_<namespace>_<resource-type>_<resource-name>_<section-name>
```

Comment:
did we considered having some placeholder for missing values? to avoid multiple _ which can be hard to read?

Reply:
You need to be explicit about resource-type. Is it the plural/camlCase...?

Reply:
Why not have the resource type before the mesh or at least before the zone?

Reply:
> did we considered having some placeholder for missing values? to avoid multiple _ which can be hard to read?

what would you use as a placeholder?

Reply:
> Why not have the resource type before the mesh or at least before the zone?

I kinda agree with is, if the resource-type is before anything else, then you can immediately tell whether to expect say an empty mesh in the case of the kri pointing to something that doesn't have a `mesh`

Reply:
Makes sense for the ordering

Reply:
> I kinda agree with is, if the resource-type is before anything else, then you can immediately tell whether to expect say an empty mesh in the case of the kri pointing to a resource type that doesn't have a mesh

Yeah, probably you're right, it's better to have the first parameter that can't be empty, i.e. `kri_zoneingress__us-east-1_kuma-system_zi-1`. I'm going to change the order then to
```
kri_<resource-type>_<mesh>_<zone>_<namespace>_<resource-name>_<section-name>
```

Reply:
Changed the order